### DISCUSSION TITLE
Use helm template --dry-run=server

### DISCUSSION THREAD

**Discussion by gakesson** [helmfile/helmfile#1131] [Ideas]
helm 3.13 introduces `--dry-run `argument for `helm template` command. Providing `--dry-run=server` means helm will access cluster for various evaluations, including the infamous helm lookup command which was extensively discussed in this [issue](https://github.com/helm/helm/issues/8137)

My suggestion is that `helmfile template` supports this argument too, but also I think that there is room for improvement in the kustomize flow. Currently a kustomized helmfile release goes through the `helm template` rendering, which means helm lookup will currently produce "wrong" result. My personal view is that helm template when used in kustomize flow should always invoke `--dry-run=server` for mutating commands like apply/sync.

Thoughts?

**Comment by isaaguilar** [comment]
see https://github.com/helmfile/helmfile/pull/1050

**Reply by yxxhero** [reply]
I will continue to do that. maybe in the next week.
